pyface/tests/test_about_dialog.py

```diff
@@ -31,29 +32,28 @@ def tearDown(self):
 
     def test_create(self):
         # test that creation and destruction works as expected
-        self.dialog._create()
-        self.event_loop()
-        self.dialog.destroy()
-        self.event_loop()
+        with self.event_loop():
+            self.dialog._create()
+        with self.event_loop():
+            self.dialog.destroy()
```

Comment:
Note that in qt to destroy widgets one needs to run the real event loop instead of simulated event processing that `self.event_loop` does. I suggest that you use https://github.com/enthought/pyface/blob/master/pyface/ui/qt4/util/gui_test_assistant.py#L98

Reply:
Recording an in-person conversation: I'm aware of the limitations here; proper clean-up should be happening in the `tearDown` method.

Many of there tests are smoke tests until we have better ways of testing the the UI is doing what we want in a cross-platform way.